Libraries/Text/TextInput/RCTBackedTextInputDelegateAdapter.m

```diff
@@ -217,7 +217,9 @@ - (BOOL)control:(NSControl *)control textView:(NSTextView *)fieldEditor doComman
     //escape
   } else if (commandSelector == @selector(cancelOperation:)) {
     [textInputDelegate textInputDidCancel];
-    [[_backedTextInputView window] makeFirstResponder:nil];
+    if (![textInputDelegate hasValidKeyDownOrValidKeyUp:@"Escape"]) {
```

Comment:
@Saadnajmi > I had another thought about this.
I think instead of this special handling here and in line 426 we could do the following - in a follow up PR

- Revert https://github.com/microsoft/react-native-macos/pull/1044
- Add a property `blurOnEscape` similar to https://reactnative.dev/docs/textinput#bluronsubmit and follow the same style
- Developers need to decide to either `blurOnEscape` or handle Escape via either validKeysDown or validKeysUp

Reply:
What do you think about that approach?
Or should we 'just' keep it as is?

Reply:
I like this idea. I think I'll leave it as a followup idea for the future, to keep this PR as is and not need a rewrite :) . Thanks for the suggestion! I think it makes great sense.

Reply:
#1419 filed.